### PR TITLE
Refactor permute!

### PR DESCRIPTION
**Pull request by bramtayl**
See #12470

### CODE REVIEW

File: base/exports.jl [hunk 1]
```diff
@@ -531,7 +531,6 @@ export
     indmax,
     indmin,
     invperm,
-    ipermute!,
```
Comment (tkelman) [inline base/exports.jl]: should be deprecated

Comment (bramtayl) [inline base/exports.jl]: how do I do that?

Comment (tkelman) [inline base/exports.jl]: see https://github.com/JuliaLang/julia/blob/master/base/deprecated.jl

Comment (bramtayl) [inline base/exports.jl]: ok i made an attempt

**Comment by bramtayl** [comment]
So this is good to go? I'm be excited to make my first successful pull request.

**Comment by StefanKarpinski** [comment]
A successful case hasn't really been made for the change – so far @ViralBShah, @JeffBezanson and I have all expressed the same sentiment in the other PR. I think and argument needs to be made for this change.

**Comment by bramtayl** [comment]
Ok. I was just looking for something easy to change, not something I feel passionate about. But:

`ipermute` is unclear, what does the i mean?
`inversepermute` is hard to read
`inverse_permute` has a _, and the typical Julia argument against _ is that it is better to have one re-factored function.

**Pull request closed without merging**